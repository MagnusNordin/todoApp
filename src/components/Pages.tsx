import styled from "styled-components"
import { useAppDispatch } from "../app/hooks"
import {
  decrementPage,
  incrementPage,
  setPaginationSize,
} from "../api/todoSlice"
import Dropdown from "react-dropdown"
import "react-dropdown/style.css"

const ItemWrapper = styled.section`
  display: flex;
  gap: 5px;
  padding-left: 5px;
  height: 24px;
  .Dropdown-root {
    font-size: 20px;
  }
`
const PageItem = styled.div`
  flex: 0 0 24px;
`

export const Pages = () => {
  const dispatch = useAppDispatch()

  const options = ["10", "20", "30"]
  const defaultOption = options[0]

  const onSelect = (value: any) => {
    console.log(value)
    const newSize: number = +value.value
    dispatch(setPaginationSize(newSize))
  }
  return (
    <ItemWrapper>
      <PageItem onClick={() => dispatch(decrementPage())}>&#x23EA;</PageItem>
      <Dropdown
        options={options}
        onChange={onSelect}
        value={defaultOption}
        placeholder=""
      />

      <PageItem onClick={() => dispatch(incrementPage())}>&#x23E9;</PageItem>
    </ItemWrapper>
  )
}
