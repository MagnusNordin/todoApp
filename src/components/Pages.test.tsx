import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { store } from "../app/store"
import { Pages } from "./Pages"

test("pages", async () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Pages />
    </Provider>,
  )

  expect(asFragment).toMatchSnapshot()
})
