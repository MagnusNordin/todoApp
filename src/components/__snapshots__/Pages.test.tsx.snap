// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pages 1`] = `[Function]`;
